src/haddock/clis/cli_analyse.py

```diff
@@ -491,11 +507,58 @@ def analyse_step(
             offline=offline,
             )
         tables = clt_table_handler(clt_file, ss_file, is_cleaned)
-        report_generator(boxes, scatters, tables, step, '.', offline)
+        report_generator(boxes, scatters, tables, step, ".", offline)
         # provide a zipped archive of the top ranked structures
         zip_top_ranked(ss_file, cluster_ranking, Path("summary.tgz"))
 
 
+def validate_format(_format: Optional[ImgFormat]) -> Optional[ImgFormat]:
+    """Validate the optional argument `format`.
+
+    Parameters
+    ----------
+    _format : Optional[ImgFormat]
+        A optential output format set by the user.
+
+    Returns
+    -------
+    Optional[ImgFormat]
+        A valid output format for the figures to be generated.
+
+    Raises
+    ------
+    ImportError
+        When the kaleido package is not installed.
+    ValueError
+        When the export format is not supported by plotly / kaleido.
+    """
+    # Optional, so if not defined, return _format (None)
+    if not _format:
+        return _format
+    # Make sure it is lower case
+    format = _format.lower()
+    # Check if part of supported output formats
+    if format in SUPPORTED_OUTPUT_FORMATS:
+        # Make sure the `kaleido` package is installed
+        try:
+            import kaleido  # noqa : F401
+        except ImportError:
+            raise ImportError(
+                f"Exporting with format {format} requires the use of `kaleido`"
+                f" package, that seems not to be installed. {os.linesep}"
+                "Please install it with: `pip install kaleido==0.2.*`"
+                )
```

Comment:
How is `kaleido` affecting the time the tests take to run?

Reply:
`kaleido` is used to generate `.png` files from plotly figures.
It takes quite some time.
Especially when generating figures with millions of data points.